backend.go

```diff
@@ -115,6 +155,34 @@ func (b *kubeAuthBackend) loadConfig(ctx context.Context, s logical.Storage) (*k
 	if config == nil {
 		return nil, errors.New("could not load backend configuration")
 	}
+
+	// Nothing more to do if loading local CA cert and JWT token is disabled.
+	if config.DisableLocalCAJwt {
+		return config, nil
+	}
+
+	// Read local JWT token unless it was not stored in config.
+	if config.TokenReviewerJWT == "" {
+		if b.localSATokenReader == nil {
+			b.localSATokenReader = newCachingFileReader(b.localJWTPath, jwtReloadPeriod, b.currentTime)
```

Comment:
The problem I have is that caller of `loadConfig()` holds read lock to `b.l` so I should not write/assign to `b.localSATokenReader` or to `b.localCACert` on line 181 below.

I think I have two options:

1. Initialize `b.localSATokenReader` and `b.localCACert` already during backend startup or at new config request.   
2. Release read lock since we know that caller holds it, acquire write lock, write `b.localSATokenReader` and `b.localCACert`, release write lock, acquire read lock again

@tomhjp @tvoran Would you have any preference which one I should do or possibly some other approach for me to try?

I think (2) sounds bit hacky to me, but on the other hand I did try to do (1) in previous version of this PR and ended up in messing up the "best effort" behavior https://github.com/hashicorp/vault-plugin-auth-kubernetes/pull/122#discussion_r759177917.  But surely, it should be doable by keeping that behavior as well :)